CI failures: "Check builds for macOS"

E.g. https://github.com/returntocorp/semgrep/runs/1420766603

Seeing:

```
 File "typing/LSP_client.ml", line 116, characters 47-55:
116 |   let json = Jsonrpc.Message.yojson_of_request json_rpc in
                                                     ^^^^^^^^
Error: This expression has type ('a, int) Stdune.either Jsonrpc.Message.t
       but an expression was expected of type
         Jsonrpc.Message.request = ProgressToken.t Jsonrpc.Message.t
       Type ('a, int) Stdune.either is not compatible with type
         ProgressToken.t = [ `Int of int | `String of string ] 
make[1]: *** [all] Error 1
make: *** [build-core] Error 2
Error: Process completed with exit code 2.
```

--- a/semgrep-core/semgrep.opam
+++ b/semgrep-core/semgrep.opam
@@ -27,7 +27,7 @@ depends: [
   "uutf"
   "re"
   "parmap"
-  "lsp"
+  "lsp" {"1.1.0"}
 ]
 
 build: [make]

--- a/semgrep-core/semgrep.opam
+++ b/semgrep-core/semgrep.opam
@@ -27,7 +27,7 @@ depends: [
   "uutf"
   "re"
   "parmap"
-  "lsp" {"1.1.0"}
+  "lsp" {= "1.1.0"}
 ]
 
 build: [make]

